api/proxy/store/eigenda_manager.go

```diff
@@ -91,18 +92,115 @@ func (m *EigenDAManager) Get(ctx context.Context,
+func (m *EigenDAManager) getEigenDAV1(
+	ctx context.Context,
+	versionedCert certs.VersionedCert,
+	opts common.GETOpts,
+) ([]byte, error) {
+	if opts.ReturnEncodedPayload {
+		return nil, fmt.Errorf("returning encoded payload is not supported for V0 certificates (EigenDA V1)")
 	}
-	if versionedCert.Version == certs.V1VersionByte && m.eigendaV2 == nil {
-		return nil, errors.New("expected EigenDA V2 backend for DA commitment type with CertV1")
+
+	verifyFnForSecondary := func(ctx context.Context, cert []byte, payload []byte, l1InclusionBlockNumber uint64) error {
+		// We don't add the cert version because EigenDA V1 only supports [certs.V0VersionByte] Certs.
+		// We also don't use the l1InclusionBlockNumber because Recency check is only supported by EigenDA V2.
+		// TODO: we should decouple the Verify function into a VerifyCert and VerifyBlob function,
+		// and only verify the cert once here, before retrievals, and then only verify the blob commitment
+		// after retrievals.
+		return m.eigenda.Verify(ctx, cert, payload)
 	}
 
-	// The eigendav2 verify function verifies the cert, but not the kzg commitment.
-	// TODO: should we be verifying those when retrieving from secondary storages like S3...?
-	verifyMethod, err := m.getVerifyMethod(versionedCert.Version)
+	var readErrors []error
+	// 1 - read payload from cache if enabled
+	// Secondary storages (cache and fallback) store payloads instead of blobs.
+	// For simplicity, we bypass secondary storages when requesting encoded payloads,
+	// since those requests are only for secure integrations and run by provers/challengers.
+	// TODO: would be nice to store blobs instead of payloads in secondary storages, such that we could standardize all
+	// storages and make them all implement the [clients.PayloadRetriever] interface.
+	// We could then get rid of the proxy notion of caches/fallbacks and only have storages.
+	if m.secondary.CachingEnabled() && !opts.ReturnEncodedPayload {
+		m.log.Debug("Retrieving payload from cached backends")
+		payload, err := m.secondary.MultiSourceRead(ctx,
+			versionedCert.SerializedCert, false, verifyFnForSecondary, opts.L1InclusionBlockNum)
+		if err == nil {
+			return payload, nil
+		}
+		m.log.Warn("Failed to read payload from cache targets", "err", err)
+		readErrors = append(readErrors, fmt.Errorf("read from cache targets: %w", err))
+	}
+
+	// 2 - read payloadOrEncodedPayload from EigenDA
+	payload, err := m.eigenda.Get(ctx, versionedCert.SerializedCert)
+	if err == nil {
+		err = m.eigenda.Verify(ctx, versionedCert.SerializedCert, payload)
+		if err != nil {
+			return nil, fmt.Errorf("verify EigenDA V1 cert: %w", err)
+		}
+		// Only backup to secondary storage if we're returning the decoded payload
+		// since the secondary stores are currently hardcoded to store payloads only.
+		// TODO: we could consider also storing encoded payloads under separate keys?
+		if m.secondary.WriteOnCacheMissEnabled() && !opts.ReturnEncodedPayload {
```

Comment:
throughout this method, you're doing checks of `ReturnEncodedPayload`, even though you explicitly error at the start if the method if this is true.

This seems like a simplification that was only partially executed?

Reply:
Yes sir! opts should not even have been passed to the L1 function! Thanks for finding these
[5b1c69f](https://github.com/Layr-Labs/eigenda/pull/1914/commits/5b1c69ffb911227dfc02b3393513dde0617110af)